core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -603,6 +605,163 @@ static HttpResponse ofFailure(Throwable cause) {
+     *
+     * @param expectedStatusClass the expected class of HTTP status
+     * @param valueTypeRef the type reference for decoded response
+     */
+    default <T> CompletableFuture<T> aggregateAs(HttpStatusClass expectedStatusClass, TypeReference<T> valueTypeRef) {
+        return aggregateAs(status -> expectedStatusClass.contains(status), valueTypeRef, defaultSubscriberExecutor());
+    }
+
+    /**
+     * Aggregates this response. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the response are received fully.
+     * The response is converted into an instance of type {@code clazz} using the default {@link ObjectMapper}
+     * and the specified {@code httpStatusFilter} decides whether the response is to be converted or not.
+     *
+     * @param httpStatusFilter the predicate which decides whether the response is to be converted or not
+     * @param clazz the class for decoded response
+     */
+    default <T> CompletableFuture<T> aggregateAs(Predicate<HttpStatus> httpStatusFilter, Class<? extends T> clazz) {
+        return aggregateAs(httpStatusFilter, clazz, defaultSubscriberExecutor());
+    }
+
+    /**
+     * Aggregates this response. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the response are received fully.
+     * The response is converted into an instance of type {@code valueTypeRef} using the default {@link ObjectMapper}
+     * and the specified {@code httpStatusFilter} decides whether the response is to be converted or not.
+     *
+     * @param httpStatusFilter the predicate which decides whether the response is to be converted or not
+     * @param valueTypeRef the type reference for decoded response
+     */
+    default <T> CompletableFuture<T> aggregateAs(Predicate<HttpStatus> httpStatusFilter, TypeReference<T> valueTypeRef) {
+        return aggregateAs(httpStatusFilter, valueTypeRef, defaultSubscriberExecutor());
+    }
+
+    /**
+     * Aggregates this response. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the response are received fully.
+     * The response is converted into an instance of type {@code clazz} using the default {@link ObjectMapper}
+     * and the specified {@code httpStatusFilter} decides whether the response is to be converted or not.
+     *
+     * @param httpStatusFilter the predicate which decides whether the response is to be converted or not
+     * @param clazz the class for decoded response
+     */
+    default <T> CompletableFuture<T> aggregateAs(Predicate<HttpStatus> httpStatusFilter,
+                                                 Class<? extends T> clazz,
+                                                 EventExecutor executor) {
+        return aggregateAs(httpStatusFilter, (content) -> {
+            try {
+                return JacksonUtil.newDefaultObjectMapper().readValue(content, clazz);
```

Comment:
`JacksonUtil.newDefaultObjectMapper()` creates new object every time. How about adding `readValue()` to `JacksonUtil` as a static method?

Reply:
I missed that point. Many thanks!